sklearn/metrics/classification.py

```diff
@@ -391,72 +392,266 @@ def jaccard_similarity_score(y_true, y_pred, normalize=True,
         over the sample set. Otherwise, return the average of Jaccard
-        similarity coefficient.
+        similarity coefficient. This is only to be specified in case
+        `average='samples'`.
+
+        .. versionchanged: 0.20
+           'normalize' is deprecated and will be removed in 0.22, instead of
+           `normalize=True` use instead just `average='samples'` and for
+           `normalize=False` use instead `average='none-samples'`.
 
     sample_weight : array-like of shape = [n_samples], optional
         Sample weights.
 
     Returns
     -------
-    score : float
-        If ``normalize == True``, return the average Jaccard similarity
-        coefficient, else it returns the sum of the Jaccard similarity
-        coefficient over the sample set.
-
-        The best performance is 1 with ``normalize == True`` and the number
-        of samples with ``normalize == False``.
+    score: float (if average is not None) or array of float, shape =\
+            [n_unique_labels]
 
     See also
     --------
     accuracy_score, hamming_loss, zero_one_loss
 
-    Notes
-    -----
-    In binary and multiclass classification, this function is equivalent
-    to the ``accuracy_score``. It differs in the multilabel classification
-    problem.
-
     References
     ----------
     .. [1] `Wikipedia entry for the Jaccard index
            <https://en.wikipedia.org/wiki/Jaccard_index>`_
 
-
     Examples
     --------
     >>> import numpy as np
     >>> from sklearn.metrics import jaccard_similarity_score
-    >>> y_pred = [0, 2, 1, 3]
-    >>> y_true = [0, 1, 2, 3]
-    >>> jaccard_similarity_score(y_true, y_pred)
-    0.5
-    >>> jaccard_similarity_score(y_true, y_pred, normalize=False)
-    2
 
     In the multilabel case with binary label indicators:
 
-    >>> jaccard_similarity_score(np.array([[0, 1], [1, 1]]),\
-        np.ones((2, 2)))
+    >>> jaccard_similarity_score(np.array([[0, 1], [1, 1]]),
+    ... np.ones((2, 2)))
     0.75
+
+    In the multiclass case:
+
+    >>> y_pred = [0, 2, 1, 3]
+    >>> y_true = [0, 1, 2, 3]
+    >>> jaccard_similarity_score(y_true, y_pred, average='macro')
+    0.5
+    >>> jaccard_similarity_score(y_true, y_pred, average='micro')
+    ... # doctest: +ELLIPSIS
+    0.333...
+
+    >>> y_pred = ['ant', 'ant', 'cat', 'cat', 'ant', 'cat']
+    >>> y_true = ['cat', 'ant', 'cat', 'cat', 'ant', 'bird']
+    >>> jaccard_similarity_score(y_true, y_pred, average='weighted')
+    ... # doctest: +ELLIPSIS
+    0.4722...
+    >>> jaccard_similarity_score(y_true, y_pred)
+    ... # doctest: +ELLIPSIS,+NORMALIZE_WHITESPACE
+    array([ 0.66...,  0. ,  0.5  ])
+    >>> jaccard_similarity_score(y_true, y_pred,
+    ... labels=['ant', 'cat', 'bird'])
+    ... # doctest: +ELLIPSIS,+NORMALIZE_WHITESPACE
+    array([ 0.66...,  0.5 ,  0.  ])
     """
 
+    average_options = (None, 'micro', 'macro', 'weighted', 'samples',
+                       'none-samples')
```

Comment:
why not include binary here?

Reply:
I find 'binary' as an average option which is does something different from other average options like 'micro', 'macro', etc, in a sense that it only applies to 'binary' input, while the other average options are defined on 'binary' as well as other types of data (technically including 'samples').

Though ofcourse that can be done, programmatic wise. Would that be preferred?

Reply:
Yes we should make it clear to users what the options are, and encourage them to investigate if they do not know what it means. Or we can show only options applicable to their y_type